src/matchengine/controller.rs

```diff
@@ -219,24 +220,28 @@ impl Controller {
         } else {
             req.limit
         };
-        let market = self
+        let markets = self
             .markets
-            .get(&req.market)
-            .ok_or_else(|| Status::invalid_argument("invalid market"))?;
-        let total_order_count = market.users.get(&req.user_id).map(|order_map| order_map.len()).unwrap_or(0);
-        let orders = market
-            .users
-            .get(&req.user_id)
-            .map(|order_map| {
-                order_map
-                    .values()
-                    .rev()
-                    .skip(req.offset as usize)
-                    .take(limit as usize)
-                    .map(|order_rc| OrderInfo::from(order_rc.deep()))
-                    .collect()
+            .iter()
+            .filter(|(key, _market)| req.market == "all" || req.market == **key)
+            .map(|(_key, market)| market);
+        let total_order_count: usize = markets
+            .clone()
```

Comment:
Will this clone trigger a lot of memory copy？

Reply:
Full order book clone ？

Reply:
Clone the iterator will not clone the HashMap